Resources/Catalog/FileCatalog.py

```diff
@@ -94,26 +129,29 @@ def w_execute( self, *parms, **kws ):
     successful = {}
     failed = {}
     failedCatalogs = []
+
     fileInfo = parms[0]
-    res = checkArgumentFormat( fileInfo )
+    res = checkArgumentFormat( fileInfo, generateMap = True )
     if not res['OK']:
       return res
-    fileInfo = res['Value']
+    fileInfo, lfnMapDict = res['Value']
+    # No need to check the LFNs again in the clients
+    kws['LFNChecking'] = False
     allLfns = fileInfo.keys()
     parms1 = parms[1:]
-    lfnsFlag = False
+
     for catalogName, oCatalog, master in self.writeCatalogs:
 
-      # Skip if metadata related method on pure File Catalog
-      if self.call in FileCatalog.write_meta_methods and not catalogName in self.metaCatalogs:
-        continue
+      # Skip if the method is not implemented in this catalog
+      if not oCatalog.hasCatalogMethod( self.call ):
```

Comment:
But it should not be since we agreed to implement on the client side all the methods as a general rule